scripts/start-utils

```diff
@@ -109,10 +117,86 @@ function log() {
   if isDebugging || isTrue "${LOG_TIMESTAMP:-false}"; then
     ts=" $(date --rfc-3339=seconds)"
   fi
-  echo "[init]${ts} $*"
+  echo -e "[init]${ts} $*"
   eval "$oldState"
 }
 
+function getCommandFromDistro(){
+  distro=$(getDistro)
+  command=
+  if [[ $distro == alpine ]]; then
+    command="su-exec"
+  else
+    command="gosu"
+  fi
+  echo $command
+}
+
+function getErrorColoredLogString() {
+  echo "\033[0;31m[ERROR] $* \033[0m"
+}
+
+function getWarningColoredLogString() {
+  echo "\033[0;33m[WARN] $* \033[0m"
+}
+
+function error() {
+  echo -e $(getErrorColoredLogString "$*")
+}
+
+function logError() {
+  log $(getErrorColoredLogString "$*")
+}
+
+function logErrorFailedTo() {
+  logError "Failed to $*"
+}
+
+function logErrorDownload() {
+  logErrorFailedTo "download $*"
+}
+
+function logErrorInstall() {
+  logErrorFailedTo "install $*"
+}
+
+function warning() {
+  echo -e $(getWarningColoredLogString "$*")
+}
+
+function logWarning() {
+  log $(getWarningColoredLogString "$*")
+}
+
+function logAutoPauseAndStopChange() {
+   logWarning "$1, set to $2 (seconds)"
+}
+
+function checkIfNumeric() {
+  [[ $1 =~ ^[0-9]+$ ]]
+}
+
+function checkIfNumericElseSetToDefault() {
+  local var_name="$1"
+  local default_value="$2"
+  
+  if ! checkIfNumeric ${!var_name} ; then
+    eval "$var_name=$default_value"
```

Comment:
Are you absolutely sure this works?

Reply:
yes tested it with multiple possibilities: 
`. "scripts/start-utils"

export TEST_VALUE="666666"

isNumericElseSetToDefault TEST_VALUE 10
checkIfNotZeroElseSetToDefault TEST_VALUE 10

export TEST_VALUE="êrer"
isNumericElseSetToDefault TEST_VALUE 10
checkIfNotZeroElseSetToDefault TEST_VALUE 10

export TEST_VALUE="6708907e"
isNumericElseSetToDefault TEST_VALUE 10
checkIfNotZeroElseSetToDefault TEST_VALUE 10

export TEST_VALUE="0"
isNumericElseSetToDefault TEST_VALUE 10
checkIfNotZeroElseSetToDefault TEST_VALUE 10

export TEST_VALUE="-1"
isNumericElseSetToDefault TEST_VALUE 10
checkIfNotZeroElseSetToDefault TEST_VALUE 10

export TEST_VALUE=9
isNumericElseSetToDefault TEST_VALUE 10
checkIfNotZeroElseSetToDefault TEST_VALUE 10
`